### PR TITLE
Update names and descriptions for services.

### PR DESCRIPTION
**Pull request by marrobi**
## What is being addressed

Update service templates  with more descriptive names, and descriptions. Part completion  of #1695 .

### CODE REVIEW

File: templates/shared_services/databricks-auth/porter.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 schemaVersion: 1.0.0
 name: tre-shared-service-databricks-private-auth
 version: 0.1.1
```
Comment (SvenAelterman) [inline templates/shared_services/databricks-auth/porter.yaml:4]: Does this version number not need to increase?

File: templates/shared_services/gitea/template_schema.json [hunk 2]
```diff
@@ -2,10 +2,32 @@
   "$schema": "http://json-schema.org/draft-07/schema",
   "$id": "https://github.com/microsoft/AzureTRE/templates/shared_services/gitea/template_schema.json",
   "type": "object",
-  "title": "Gitea Shared Service",
-  "description": "Provides Gitea shared service",
+  "title": "Gitea - Git Mirror",
+  "description": "The Gitea shared service is typically used for mirroring external Git repositories.",
   "required": [],
-  "properties": {},
+  "properties": {
```
Comment (SvenAelterman) [inline templates/shared_services/gitea/template_schema.json:8]: I don't understand why there weren't already here?

Comment (SvenAelterman) [reply templates/shared_services/gitea/template_schema.json:8]: And same for some of the other services.

File: templates/shared_services/databricks-auth/porter.yaml (hunk 1 above)
Comment (marrobi) [reply templates/shared_services/databricks-auth/porter.yaml:4]: Thanks.

File: templates/shared_services/gitea/template_schema.json (hunk 2 above)
Comment (marrobi) [reply templates/shared_services/gitea/template_schema.json:8]: They exist as default properties in `api_app/schemas/workspace_service.json`` as every service needs them. We didn't forsee the need to set default values. We shoudl probably lint to ensure they exist for services as in time they need adding to them all.

Comment (marrobi) [reply templates/shared_services/gitea/template_schema.json:8]: Issue https://github.com/microsoft/AzureTRE/issues/1695 covers

**Comment by marrobi** [comment]
/test-force-approve

Passed: https://github.com/microsoft/AzureTRE/actions/runs/5737334132

**Pull request merged**